### PR TITLE
precompiler flag to switch data storage type to float

### PR DESCRIPTION
**Pull request by Jojosito**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -11,6 +11,14 @@ if(DOXYGEN_FOUND)
                       COMMENT "Generating API documentation with Doxygen" VERBATIM)
 endif(DOXYGEN_FOUND)
 
+# DataPoint type flag
+if (DATA_POINT_TYPE STREQUAL "float")
```
Comment (greenwald) [inline CMakeLists.txt]: `DATA_POINT_TYPE` -> `YAP_DATA_POINT_TYPE`

File: CMakeLists.txt [hunk 2]
```diff
@@ -11,6 +11,14 @@ if(DOXYGEN_FOUND)
...
+	message("Using float as type for DataPoint")
+	set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -D DATA_POINT_FLOAT")
```
Comment (greenwald) [inline CMakeLists.txt]: `DATA_POINT_FLOAT` -> `YAP_DATA_POINT_FLOAT`

File: include/DataPoint.h [hunk 3]
```diff
@@ -28,6 +28,13 @@
 
 namespace yap {
 
+/// precompiler flag for data storage type
+#ifdef DATA_POINT_FLOAT
```
Comment (greenwald) [inline include/DataPoint.h]: `DATA_POINT_FLOAT` -> `YAP_DATA_POINT_FLOAT`

And we should make this a member typedef of `DataPoint` simply called `type`

File: include/DataPoint.h [hunk 4]
```diff
@@ -61,9 +68,9 @@ class DataPoint
 
     /// Data storage for all DataAccessors
     /// first index is for the DataAccessor
-    /// second index is for the symmeterization state (as known by the DataAccessor)
-    /// third index is internal to the DataAccessor
-    std::vector<std::vector<double> > Data_;
+    /// second index is for the symmeterization state (as known by the DataAccessor) internal to the DataAccessor
+    /// i = symIndex * size + position
+    std::vector<std::vector<data_point_type> > Data_;
```
Comment (greenwald) [inline include/DataPoint.h]: `data_point_type` --> `type`

**Comment by Jojosito** [comment]
I tried to change the type to float in #122 . The tests still fail. When I store the FourMomenta as double, and the rest as float, the tests pass.

**Comment by Jojosito** [comment]
Found the cause of the fail -1.0 == Approx( 0.0 )

In HelicityAngles.cxx, line 92, epsilon is based on double, but should be float (if the FourMomenta are actually floats being casted to doubles)

File: include/DataPoint.h [hunk 5]
```diff
@@ -59,11 +59,18 @@ class DataPoint
 
 private:
 
+/// precompiler flag for data storage type
```
Comment (greenwald) [inline include/DataPoint.h]: I think the type should be public. It's potentially useful to have access to it.

**Pull request merged**